### ISSUE TITLE
Documentation for "..." in projection

### ISSUE THREAD

**Issue by manor** [erezsh/Preql#22]
I get why this:

`
join(c: Country.name, n:["Palau", "Nauru"].item) {...c}
`

returns this:

![image](https://user-images.githubusercontent.com/596326/113526050-550cf180-9586-11eb-996f-873630a3bc3f.png)

and even why this:

`
join(c: Country.name, n:["Palau", "Nauru"].item) {...n}
`

returns this:

![image](https://user-images.githubusercontent.com/596326/113526068-75d54700-9586-11eb-8fd1-8d7bff3d8fa6.png)

but what does this even mean?

`
join(c: Country.name, n:["Palau", "Nauru"].item) {...}
`

because this is what it returns:

![image](https://user-images.githubusercontent.com/596326/113526090-930a1580-9586-11eb-8997-e0145cc98c80.png)

In the context of a project, '...' means all the columns, but the type of the result of the join and/or the meaning of "..." are a bit unclear. Also, I understand rows and tables but it looks like entire structs are being returned in the last case:

![image](https://user-images.githubusercontent.com/596326/113526177-ef6d3500-9586-11eb-9a90-4c3e4f8dc233.png)

**Comment by erezsh** [comment]
Yes, the `join()` operation returns a struct for each table. Joining two tables creates two structs.

From the docs (https://preql.readthedocs.io/en/latest/preql-modules.html#join):

```
Returns: A new table, where each column is a struct representing one of the joined tables.
```

When you do `join_result{...n}`, you are telling Preql to inline the struct called `n`. You could also do `join_result{n.item}`, with a similar result in that case.

In Preql, `my_table {...}` is essentially a no-op, which just returns all the columns. If they are structs, they will remain structs.

Is it clear now?

If you have suggestions on how to improve the docs or tutorial, I'll be happy to hear them.
